src/libsystemd/sd-netlink/netlink-types.c

```diff
@@ -391,6 +395,7 @@ static const char* const nl_union_link_info_data_table[] = {
         [NL_UNION_LINK_INFO_DATA_XFRM] = "xfrm",
         [NL_UNION_LINK_INFO_DATA_IFB] = "ifb",
         [NL_UNION_LINK_INFO_DATA_BAREUDP] = "bareudp",
+        [NL_UNION_LINK_INFO_DATA_BATADV] = "batadv",
```

Comment:
Seems not used...

Reply:
Yes, it's unused at the moment. But it will be used in the future when B.A.T.M.A.N. moves the routing_algo selection to RTNL. @ecsv can maybe comment further on that.

Should it be removed in the meanwhile?

Reply:
I would also propose to remove it for now. I first have to play around with the concept of using the link_info_data for the transport of "early" configuration in batman-adv (like the infamous routing_algo)

Reply:
When I remove this the tests fail with:
```
ip6tnl_mode: 2 â
--- stderr ---
Assertion '!(i >= 0 && i < size ? sparse ? rev != i && rev != -1 : val == NULL || rev != i : val != NULL || rev != -1)' failed at src/shared/test-tables.h:34, function _test_table(). Aborting.
-------

```
not really sure why this happens

Reply:
So there is no need to remove this anymore as @ecsv just submitted a patch to BATMAN to control the routing algo over RTNL.

Reply:
OK. So maybe `rtnl_link_info_data_batadv_types` needs to be updated?